inginious/frontend/plugins/multilang/problems/custom_input_notebook.py

```diff
@@ -2,57 +2,79 @@
 import web
 
 from inginious.frontend.plugins.utils import get_mandatory_parameter
-from inginious.client.client_sync import ClientSync
 from inginious.frontend.pages.api._api_page import APIAuthenticatedPage
 from inginious.frontend.parsable_text import ParsableText
 
 
-def custom_input_notebook(client):
+def custom_input_notebook(client, custom_test_manager):
     """
-    This function returns a CustomInputNotebookManager in charge of running the selected tests by the student to run.
+    This function returns a CustomTestNotebookAPI in charge of running the selected tests by the student to run.
     This is instead of doing a new submission whenever the student wants to check the result of some few tests.
     :param client:
-    :return: CustomInputNotebookManager
+    :param custom_test_manager:
+    :return: CustomTestNotebookAPI
     """
 
-    class CustomInputNotebookManager(APIAuthenticatedPage):
+    class CustomTestNotebookAPI(APIAuthenticatedPage):
         def __init__(self):
             self._client = client
+            self._custom_test_manager = custom_test_manager
 
-        def add_unsaved_job(self, task, inputdata):
-            temp_client = ClientSync(self._client)
-            return temp_client.new_job(task, inputdata,
-                                       "Custom input notebook - " + self.user_manager.session_username())
+        def _add_custom_test_job(self, task, input_data):
+            return self._custom_test_manager.new_job(task, input_data,
+                                                     "Custom test notebook - " + self.user_manager.session_username())
 
-        def is_valid_input(self, problem_id, user_input):
+        def _is_valid_input(self, problem_id, user_input):
             if (problem_id + "/input") not in user_input:
                 return False
 
             if not user_input[problem_id + "/input"]:
                 return False
 
             # The selected tests are given as a string separated by comma
-            tests = user_input[problem_id + "/input"].split(',')
+            tests = user_input[problem_id + "/input"].split(",")
             if len(tests) <= 0 or len(tests) > 3:
```

Comment:
We know the meaning of these 0 and 3, but it's better let them clear with a variable, something like `max_num_of_tests = 3`.